### ISSUE TITLE
todo in erlang_ty_to_ast

### ISSUE THREAD

**Issue by skogsbaer** [etylizer/etylizer#160]
```erlang
-module(test).

-compile(export_all).
-compile(nowarn_export_all).

-type exp() :: integer() | {add, exp(), exp()} | {uminus, exp()}.

-spec plus(integer(), integer()) -> integer().
plus(X, Y) -> X + Y.

-spec eval(exp()) -> integer().
eval(E) ->
    case E of
        {add, E1, E2} -> plus(eval(E1), eval(E2));
        %{uminus, E} -> eval(E);
        _ -> 1
    end.
```

Error:

```
escript: exception error: todo
  in function  ast_lib:erlang_ty_to_ast/2 (/Users/swehr/devel/etylizer/src/ast_lib.erl, line 203)
  in call from lists:map_1/2 (lists.erl, line 2082)
  in call from lists:map/2 (lists.erl, line 2077)
  in call from ast_lib:'-erlang_ty_to_ast/2-fun-3-'/2 (/Users/swehr/devel/etylizer/src/ast_lib.erl, line 152)
  in call from dnf_ty_tuple:'-transform/2-lc$^1/1-0-'/2 (/Users/swehr/devel/etylizer/src/erlang_types/bdd.hrl, line 207)
  in call from dnf_ty_tuple:'-transform/2-fun-3-'/6 (/Users/swehr/devel/etylizer/src/erlang_types/bdd.hrl, line 207)
  in call from dnf_ty_tuple:'-transform/2-fun-0-'/3 (/Users/swehr/devel/etylizer/src/erlang_types/bdd.hrl, line 211)
  in call from dnf_var_ty_tuple:'-transform/2-fun-3-'/6 (/Users/swehr/devel/etylizer/src/erlang_types/bdd.hrl, line 206)
```

**Comment by albsch** [comment]
Smaller example with same error:

```erlang
-module(test).
-compile([export_all, nowarn_export_all]).
-spec eval({a} | {}) -> any().
eval({E}) -> eval(E).
```

**Comment by albsch** [comment]
This test crashes now with out of memory exception with #165 ... still a performance issue left.

**Comment by skogsbaer** [comment]
Fixed by #165
